app/pages/admin/project-status.jsx

```diff
@@ -76,10 +79,28 @@ class ProjectStatus extends Component {
   handleToggle(event, workflow) {
     this.setState({ error: null });
     const checked = event.target.checked;
+    const defaultWorkflowId = this.state.project.configuration.default_workflow;
+
+    const promises = [this.getWorkflows(), this.getProject()];
+
+    if (defaultWorkflowId === workflow.id && workflow.active) {
+      Dialog.alert(
+        <WorkflowDefaultDialog closeButton={true} required={true} />
+      )
+      .then(() => {
+        Promise.all(promises)
+          .then(() => {
+            if (!this.state.error) {
+              workflow.update({ active: checked }).save();
+              this.state.project.update({ 'configuration.default_workflow': undefined }).save();
+            }
+          })
+          .catch(error => this.setState({ error }));
+      })
+      .catch(error => this.setState({ error }));
+    }
 
-    return workflow.update({ 'active': checked }).save()
-      .then(() => this.getWorkflows())
-      .catch(error => this.setState({ error }))
+    workflow.update({ active: checked }).save();
```

Comment:
Also related to the question in the PR: `if (defaultWorkflowId !== workflow.id)` then update the workflow. We don't want to always update the active state for the workflow.